src/Sylius/Bundle/ApiBundle/Validator/GatewayConfig/GatewayConfigValidator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\GatewayConfig;
+
+use Sylius\Bundle\ApiBundle\Validator\Constraints\GatewayConfig;
+use Sylius\Bundle\PayumBundle\Model\GatewayConfigInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class GatewayConfigValidator extends ConstraintValidator
+{
+    public const OFFLINE_GATEWAY_FACTORY = 'offline';
+
+    /** @param array<string, string> $factoryNames */
+    public function __construct(private array $factoryNames)
+    {
+    }
+
+    /** @param GatewayConfigInterface $value */
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof GatewayConfig) {
+            throw new UnexpectedTypeException($constraint, GatewayConfig::class);
+        }
+
+        if (!$value instanceof GatewayConfigInterface) {
+            throw new UnexpectedValueException($value, GatewayConfigInterface::class);
+        }
```

Comment:
Would be cool to place `$value` check first.

```suggestion
        if (!$value instanceof GatewayConfigInterface) {
            throw new UnexpectedValueException($value, GatewayConfigInterface::class);
        }
        if (!$constraint instanceof GatewayConfig) {
            throw new UnexpectedTypeException($constraint, GatewayConfig::class);
        }
```

Reply:
Why? 🤔

Reply:
For the same reason why we sort DI services, api groups alphabetically or classes constructor arguments and classes properties in the same way -- easy of read :slightly_smiling_face: 

`$value` appears first, then it should be validated first. 

![image](https://github.com/Sylius/Sylius/assets/870747/d9c6e551-dce1-46ec-b269-ae862634c104)
![image](https://github.com/Sylius/Sylius/assets/870747/5cb240ae-0104-4072-acb2-b10407fbd443)
![image](https://github.com/Sylius/Sylius/assets/870747/a62df0de-460a-4717-8f93-5ec96c65f02f)
![image](https://github.com/Sylius/Sylius/assets/870747/9c1ae615-85b1-45f7-8ae5-78d6e3414261)
![image](https://github.com/Sylius/Sylius/assets/870747/c522825c-8ff4-468b-badd-43b5d8c3738a)